Realm/RLMQueryUtil.mm

```diff
@@ -113,6 +141,68 @@ void set_base_table(const Table*) override {}
     }
 };
 
+
+// Equal and ContainsSubstring are used by QueryBuilder::add_string_constraint as the comparator
+// for performing diacritic-insensitive comparisons.
+
+template <NSUInteger options>
+struct Equal {
+    using CaseSensitive = Equal<options & ~NSCaseInsensitiveSearch>;
```

Comment:
I'm not sure these aliases are actually useful. They look weird on their own (my initial thought before seeing how they're used was "why would you do `Equal<NSDiacriticInsensitivePredicateOption>::CaseInsensitive` rather than `Equal<NSDiacriticInsensitivePredicateOption|NSCaseInsensitiveSearch>`") and t he fact that they're defined in two places but only used in one means they aren't reducing duplication